client/blocks/stats-navigation/index.js

```diff
@@ -83,11 +83,34 @@ class StatsNavigation extends Component {
 				};
 			} )
 		),
+		availableModuleToggles: [],
 	};
 
 	static getDerivedStateFromProps( nextProps, prevState ) {
-		if ( prevState.pageModules !== nextProps.pageModuleToggles ) {
-			return { pageModules: nextProps.pageModuleToggles };
+		const availableModuleToggles = ( AVAILABLE_PAGE_MODULES[ nextProps.selectedItem ] || [] ).map(
+			( toggleItem ) => {
+				// disable the "videos" toggle on sites that do not have VideoPress enabled
+				// the toggle will be disabled (grayed out and non interactive)
+				const shouldDisableVideoToggle = ! nextProps.hasVideoPress && toggleItem.key === 'videos';
+
+				return {
+					...toggleItem,
+					disabled: shouldDisableVideoToggle,
+					defaultValue: shouldDisableVideoToggle === false,
+				};
+			}
+		);
+
+		// toggle the visibility of video module itself
+		if ( ! nextProps.hasVideoPress ) {
+			nextProps.pageModuleToggles.videos = false;
+		}
```

Comment:
Moved the toggles logic into the `getDerivedStateFromProps` function.

And centralized the logic behind toggling the video module itself.